agent.go

```diff
@@ -242,11 +249,16 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 		portmax:     config.PortMax,
 		trickle:     config.Trickle,
 		log:         loggerFactory.NewLogger("ice"),
+		net:         config.Net,
 
 		forceCandidateContact: make(chan bool, 1),
 	}
 	a.haveStarted.Store(false)
 
+	if a.net == nil {
+		a.net = vnet.NewNet(nil)
+	}
```

Comment:
Let's add a warning log if a.net is not nil.
See: https://github.com/pion/transport/wiki/vnet:-Usage-Policy#log-warning-from-each-pion-module-if-vnet-is-enabled

Reply:
done!